spec/controllers/idv/confirmations_controller_spec.rb

```diff
@@ -25,34 +27,60 @@
     end
 
     describe 'all questions answered correctly' do
-      it 'shows success' do
+      it 'updates profile and redirects to original SAML Authn request' do
+        user.identities.create(
+          last_authenticated_at: Time.current,
+          service_provider: sp1_saml_settings.issuer
+        )
+        subject.session[:saml_request_url] = sp1_authnrequest
         complete_idv_session(true)
 
         get :index
 
-        expect(response.status).to eq 200
-        expect(response.body).to include(t('idv.titles.complete'))
+        expect(response).to redirect_to subject.session[:saml_request_url]
+        expect(flash[:success]).to eq(t('idv.titles.complete'))
 
         profile.reload
 
         expect(profile).to be_active
         expect(profile).to be_verified
+
+        # PII removed from session
+        expect(subject.user_session[:idv][:applicant]).to eq nil
+      end
+
+      it 'updates profile and redirects to IdP profile when original SAML Authn request invalid' do
```

Comment:
If the SAML authn request is invalid shouldn't we detect that right away before the user goes through login + IdV?  Not a bit dial I guess because invalid request should never happen.